pdns/speedtest.cc

```diff
@@ -862,6 +863,100 @@ struct NSEC3HashTest
   DNSName d_name = DNSName("www.example.com");
 };
 
+struct ReadWriteLockSharedTest
+{
+  explicit ReadWriteLockSharedTest(ReadWriteLock& lock): d_lock(lock)
+  {
+  }
+
+  string getName() const { return "RW lock shared"; }
+
+  void operator()() const {
+    for (size_t idx = 0; idx < 1000; ) {
+      ReadLock wl(d_lock);
+      ++idx;
+    }
+  }
+
+private:
+  ReadWriteLock& d_lock;
+};
+
+struct ReadWriteLockExclusiveTest
+{
+  explicit ReadWriteLockExclusiveTest(ReadWriteLock& lock): d_lock(lock)
+  {
+  }
+
+  string getName() const { return "RW lock exclusive"; }
+
+  void operator()() const {
+    for (size_t idx = 0; idx < 1000; ) {
```

Comment:
please put the increment in `for (...)`

Reply:
I'm worried doing so would cause the compiler to try to elude the actual locking part.

Reply:
Semantically it's 100% equal, so if once could be optimized away, the other could too...

Reply:
OK, I wasn't sure the for iteration statement was considered part of the body, and thus would the lock object would still exist at that point.